app/dependencies.py

```diff
@@ -286,24 +286,70 @@ async def get_current_user(
 
 
 def is_user(
-    user: models_core.CoreUser = Depends(
-        get_user_from_token_with_scopes([[ScopeType.API]]),
-    ),
-) -> models_core.CoreUser:
+    excluded_groups: list[GroupType] | None = None,
+    included_groups: list[GroupType] | None = None,
+    excluded_account_types: list[AccountType] | None = None,
+    included_account_types: list[AccountType] | None = None,
+    exclude_external: bool = False,
+) -> Callable[[models_core.CoreUser], models_core.CoreUser]:
     """
     A dependency that will:
         * check if the request header contains a valid API JWT token (a token that can be used to call endpoints from the API)
         * make sure the user making the request exists
 
     To check if the user is not external, use is_user_a_member dependency
-    To check if the user is not external and is the member of a group, use is_user_a_member_of generator
+    To check if the user is not external and is the member of a group, use is_user_in generator
+    To check if the user has an ecl account type, use is_user_an_ecl_member dependency
     """
-    return user
+
+    excluded_groups = excluded_groups or []
+    excluded_account_types = excluded_account_types or []
+    included_account_types = included_account_types or list(AccountType)
+
+    def is_user(
```

Comment:
Three cases remains untested, would you mind adding a test for them?
- `excluded_account_types`
- `is_user_external`
- `included_account_types`